src/cbor/common.c

```diff
@@ -158,3 +158,101 @@ cbor_item_t *cbor_move(cbor_item_t *item) {
   item->refcount--;
   return item;
 }
+
+bool cbor_equal(cbor_item_t *item1, cbor_item_t *item2) {
+  size_t i;
+  cbor_item_t * cur_item1, * cur_item2;
+  bool ret;
+  
+  if (item1 != NULL && item2 != NULL) {
+    if (item1->type != item2->type) {
+      return false;
+    } else {
+      switch (item1->type) {
+        case CBOR_TYPE_UINT:
+        case CBOR_TYPE_NEGINT:
```

Comment:
This will erroneously return true for an int and a negint with the same data. It should also definitely be covered by a test.

Reply:
No because just [above](https://github.com/PJK/libcbor/blob/46f79c7087891458a5d30c2c48fe831d82a1f3f9/src/cbor/common.c#L168), there is a preliminary check:

```C
    if (item1->type != item2->type) {
      return false;
    } else {
      switch (item1->type) {
        case CBOR_TYPE_UINT:
        case CBOR_TYPE_NEGINT:
```

So if the types are different, items are considered different anyway.

Reply:
You are right, sorry